### DISCUSSION TITLE
Can you cancel or reset a delay using an event while it's running?

### DISCUSSION THREAD

**Discussion by mcntsh** [statelyai/xstate#3578] [Q&A] [answered]
Is this possible to change or cancel a running delay using an event,? Here's a simplified example of what I'm trying to do:
 
 ```javascript
createMachine(
  {
    id: 'toggle',
    initial: 'inactive',
    states: {
      active: {
        on: {
          TOGGLE: 'inactive',
          LENGTHEN_DELAY: {
            actions: (context) => (context.delay = 60_000),
          },
        },
        after: [
          {
            delay: 'someDelay',
            target: 'inactive',
          },
        ],
      },
      inactive: { on: { TOGGLE: 'active' } },
    },
  },
  {
    delays: {
      someDelay: (context) => context.delay || 10_000,
    },
  }
)
```

If my machine gets a `LENGTHEN_DELAY` event then I want to extend the running delay from 10 seconds to 60.

**Comment by mcntsh** [comment, marked as answer]
[There's a way to cancel the delay with the `cancel` action creator](https://xstate.js.org/docs/guides/delays.html#delayed-events)

**Comment by Andarist** [comment]
Note that you shouldn't mutate the `context` directly in your transition for the `LENGTHEN_DELAY` event. `assign` should be used to do that in an immutable manner.
